src/components/wallet/methods/updateAccount.ts

```diff
@@ -8,18 +8,34 @@ import { handleError } from '@services/error'
 import { Wallet } from '../wallet'
 
 export default async function updateAccount(this: Wallet) {
+  function cbQuery({ this: wallet }) {
+    wallet.server
```

Comment:
So, I see you're using the old style of javascript of chaining callbacks. Very understandable, I did that too when I first started working on Michael's JS projects.

What I learned is that ES6 added a bunch of native async capabilities that make these much more readable.

For example, your current `wallet.server.call()` code can be converted to this:
```
let resp = await wallet.server.claimableBalances().claimant(wallet.account.publicKey).call()
```
You can then use `resp` on the next line like so:
```
resp.records.forEach((record) => {
    wallet.logger.response(...)
})
```
Essentially, you can get rid of the deep nesting of javascript code within other function calls. It makes JS code so much more readable, so please use this style from hereon out!

Reply:
Check out the refactor here https://github.com/stellar/stellar-demo-wallet/pull/39/commits/fabd6cc73f8e791ca2d2eb64a8692c7951363c20